### DISCUSSION TITLE
Build for Kotlin Multiplatform

### DISCUSSION THREAD

**Discussion by Otlichnick** [trustwallet/wallet-core#2817] [Q&A] [answered]
Is there any way to build wallet core in shared module of kmm and use it as Kotlin Multiplatform library?

**Comment by Milerius** [comment]
An issue Is created: https://github.com/trustwallet/wallet-core/issues/2740

I believe @MaximPestryakov did some research on the topic

**Comment by MaximPestryakov** [comment, marked as answer]
Hello @Otlichnick 

Right now there is no convenient way to use Wallet Core in KMM project. We are working on it. But despite this you can add Wallet Core dependencies in KMM project

For Android:
```kotlin
val androidMain by getting {
    dependencies {
        implementation("com.trustwallet:wallet-core:3.1.7")
    }
}

```

For iOS using CocoaPods Kotlin plugin:
```kotlin
plugins {
    kotlin("native.cocoapods")
}

cocoapods {
    pod(name = "TrustWalletCore", version = "3.1.7", moduleName = "WalletCore")
}
```

**Reply by Otlichnick** [reply]
Yes, but the goal is incapsulate WalletCore inside shared module, add some extra functionality and connect to the existing iOS and Android applications.
